### PR TITLE
Add warning against using response caching for authenticated content

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes: https://github.com/aspnet/Docs/issues/3044

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
ping

File: aspnetcore/performance/caching/response.md [hunk 1]
```diff
@@ -35,6 +35,9 @@ The web server can cache responses by adding the response caching middleware. Se
 
 The `ResponseCacheAttribute` specifies the parameters necessary for setting appropriate headers in response caching. See [ResponseCacheAttribute](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute)  for a description of the parameters.
 
+>[!WARNING]
+> Remember to disable caching for content that may contain information for authenticated clients. Caching should only be used for content available to anyone.
```
Comment (Rick-Anderson) [inline aspnetcore/performance/caching/response.md]: ~~Remember to~~ disable caching for content that may contain information for authenticated clients.

Comment (Rick-Anderson) [inline aspnetcore/performance/caching/response.md]: Would it be more clear to say something like:

Caching should only be used for content that contains no sensitive information.

File: aspnetcore/performance/caching/middleware.md [hunk 2]
```diff
@@ -33,6 +33,9 @@ Configure the application to use the middleware when processing requests. The sa
 
 The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
 
+>[!WARNING]
+> Responses containing content for authenticated clients must be marked as not cacheable to prevent the middleware from storing and serving those responses.
```
Comment (blowdart) [inline aspnetcore/performance/caching/middleware.md]: You should explain *how* here.

File: aspnetcore/performance/caching/response.md (hunk 1 above)
Comment (blowdart) [inline aspnetcore/performance/caching/response.md]: "may contain information" should change to "contains".

"Caching should only be used for content available to anyone" should change to "Caching should only be used for content that does not change based on a user's identity, or on whether a user is logged in or not."

**Comment by JunTaoLuo** [comment]
:up::date:

**Comment by blowdart** [comment]
LGTM

File: aspnetcore/performance/caching/middleware.md [hunk 3]
```diff
@@ -33,6 +33,9 @@ Configure the application to use the middleware when processing requests. The sa
 
 The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
 
+>[!WARNING]
+> Responses containing content for authenticated clients must be marked as not cacheable to prevent the middleware from storing and serving those responses. See [Conditions for caching](#conditions-for-caching) for details on how the middleware determine if a response is cacheable.
```
Comment (Eilon) [inline aspnetcore/performance/caching/middleware.md]: Typo: `how the middleware determine` should be `how the middleware determine*s*`.

Comment (JunTaoLuo) [inline aspnetcore/performance/caching/middleware.md]: Oops. @Rick-Anderson can you include this fix in another PR. Probably not worth an entire commit/PR to correct.

Comment (Rick-Anderson) [inline aspnetcore/performance/caching/middleware.md]: @JunTaoLuo  NP

Comment (Rick-Anderson) [inline aspnetcore/performance/caching/middleware.md]: @JunTaoLuo  FIXED in https://github.com/aspnet/Docs/commit/600c1afc4b6bdb2915eed80f9fdb8bb2fa731b17

Comment (JunTaoLuo) [inline aspnetcore/performance/caching/middleware.md]: Thanks!

Comment (Eilon) [inline aspnetcore/performance/caching/middleware.md]: Thanks guys 😄

**Pull request merged**